source/globalCommands.py

```diff
@@ -2391,3 +2394,113 @@ def script_recognizeWithUwpOcr(self, gesture):
 #: The single global commands instance.
 #: @type: L{GlobalCommands}
 commands = GlobalCommands()
+
+class ConfigProfileActivationCommands(ScriptableObject):
+	"""Singleton scriptable object that collects scripts for available configuration profiles."""
+
+	scriptCategory = SCRCAT_CONFIG_PROFILES
+
+	@classmethod
+	def __new__(cls, *args, **kwargs):
+		# Iterate through the available profiles, creating scripts for them.
+		for profile in config.conf.listProfiles():
+			cls.addScriptForProfile(profile)
+		return 		super(ConfigProfileActivationCommands, cls).__new__(cls)
+
+	@classmethod
+	def _getScriptNameForProfile(cls, name):
+		invalidChars = set()
+		for c in name:
+			if not c.isalnum() and c != "_":
+				invalidChars.add(c)
+		for c in invalidChars:
+			name=name.replace(c,'_')

```

Comment:
This will increase the chances of a name collision. Consider two profile names `chrome^` and `chrome*`. How are name collisions detected and handled?

I suggest looking to see if the name is already used and append an int ID.

Reply:
Given you need the name created to reproducible every time so you can find it again in the `removeScriptForProfile` function, you could just append a base16 encoding of the string (from the value prior to substitutions). If you want to reduce the length of this string, increase the size of the base encoding (up to the number of valid characters).